java/src/ApiImpl.java

```diff
@@ -15,27 +18,77 @@ public abstract class ApiImpl implements AuthenticatedApi{
     private Vault vault;
 
     public List<Reservation> getMyReservations(int userId){
-        //TODO
-		return Collections.emptyList();
+        return ((List<Object[]>)database.runQuery("select * from reservations where user_id = " + userId)).stream().map(Reservation::map).collect(Collectors.toList());
     }
 
+    @AuthorizedRoles(value = {AuthorizedRoles.Roles.ADMIN})
     public List<Invoice> getUserInvoices(String userId){
-		//TODO        
-		return Collections.emptyList();
+        return database.createPreparedStatement("select * from invoices where user_id = " + userId, Invoice.class).getResultList();
     }
 
     public void setMyAlternatePhoneNumbers(String commaDelimitedNumbers){
-        //TODO
+        String[] numbers = commaDelimitedNumbers.split(",");
+        for (String number : numbers) {
+            if(number.length() < 8){
+                throw new WebApplicationException("Number too short", 400);
+            }
+            database.createPreparedStatement("insert into user_contact (user_id, tlf) values (?1,?2)", Void.class)
+                    .setParameter(1, getLoggedInUser())
+                    .setParameter(2, number)
+                    .executeUpdate();
+        }
+        database.createPreparedStatement("delete from user_contact where user_id = ?1 and tlf not in ("+commaDelimitedNumbers+")", Void.class)
+                .setParameter(1, getLoggedInUser())
+                .executeUpdate();
     }
 
     public void updateMyEmail(UserObject userData){
-        //TODO;
+        database.persist(userData);
+    }
+
+    public String doTheStuff(String userDefined){
```

Comment:
I'm having trouble understanding this code. Can you maybe change the name and variable names, or rewrite the whole thing? Maybe with comments?